app/src/main/kotlin/com/wire/android/ui/calling/SharedCallingViewModel.kt

```diff
@@ -247,25 +253,37 @@ class SharedCallingViewModel @Inject constructor(
 
     fun toggleSpeaker() {
         viewModelScope.launch {
-            callState = if (callState.isSpeakerOn) {
+            if (callState.isSpeakerOn) {
+                callState = callState.copy(isSpeakerOn = false)
                 turnLoudSpeakerOff()
-                callState.copy(isSpeakerOn = false)
             } else {
+                callState = callState.copy(isSpeakerOn = true)
                 turnLoudSpeakerOn()
-                callState.copy(isSpeakerOn = true)
+            }
+        }
+    }
+
+    fun flipCamera() {
+        viewModelScope.launch {
+            if (callState.isOnFrontCamera) {
+                callState = callState.copy(isOnFrontCamera = false)
+                flipToBackCamera(conversationId)
+            } else {
+                callState = callState.copy(isOnFrontCamera = true)
+                flipToFrontCamera(conversationId)
```

Comment:
if `flipToFrontCamera(conversationId)` failed for any reason, it will create an inconsistency between the state object and the actual camera, since the state is updated regardless whether the use case succeeded or not

Reply:
Nope, if the back camera is not available then a black screen  or current frozen video stream will be displayed. 
I did it in this way to avoid the delay that could between when tapping the icon and flipping the camera.